cmd/outline-ss-server/config_example.yml

```diff
@@ -4,6 +4,10 @@ services:
         address: "tcp://[::]:9000"
       - type: direct
         address: "udp://[::]:9000"
+      - type: proxy_protocol
```

Comment:
Something doesn't feel right. A "PROXY"  listener should be able to take another listener as input, but that is not possible in this format. I mean, it kind of is, the "address" just means "direct".

Something more "correct" would be:
```yml
- listeners:
  - type: proxy_protocol
    listener:
      type: direct
      address: "tcp://[::]:9000"
```

That allows us to define consistent types that can be nested.

Note that we are wasting the "direct" type. We might as well have something like

```yml
- listeners:
  - type: proxy_protocol
    listener:
      type: tcp
      address: "[::]:9000"
```

This allows use to define what each listener can take (e.g. unix takes a path, not a net address, a tcp listener could bind to an interface).

In terms of representation, an alternative to using a `type` field is to use the field names for type selection:

```yml
- listeners:
  - proxy_protocol:
      listener:
        tcp: {address: "[::]:9000"}
```

Not sure which approach is better. Caddy has both.